Jenkinsfile

```diff
@@ -0,0 +1,253 @@
+        BRANCH_TEST = "release"
+        BRANCH_PROD = "master"
+
+        DEPLOY_DEV = "dev"
+        DEPLOY_TEST = "test"
+        DEPLOY_PROD = "beta"
+
+        GIT_TYPE = "Github"
+        GIT_CREDS = "github-bi-user"
+        GITLAB_CREDS = "bi-gitlab-id"
+
+        ORGANIZATION = "ons"
+        TEAM = "bi"
+        MODULE_NAME = "business-index-api"
+    }
+    options {
+        skipDefaultCheckout()
+        buildDiscarder(logRotator(numToKeepStr: '30', artifactNumToKeepStr: '30'))
+        timeout(time: 30, unit: 'MINUTES')
+        timestamps()
+    }
+    agent any
+    stages {
+        stage('Checkout'){
+            agent any
+            steps{
+                deleteDir()
+                checkout scm
+                stash name: 'app'
+                sh "$SBT version"
+                script {
+                    version = '1.0.' + env.BUILD_NUMBER
+                    currentBuild.displayName = version
+                    env.NODE_STAGE = "Checkout"
+                }
+            }
+        }
+
+        stage('Build'){
+            agent any
+            steps {
+                colourText("info", "Building ${env.BUILD_ID} on ${env.JENKINS_URL} from branch ${env.BRANCH_NAME}")
+                script {
+                    env.NODE_STAGE = "Build"
+                    sh '''
+                        $SBT clean compile "project api" universal:packageBin coverage test coverageReport
+                    '''
+                    stash name: 'compiled'
+                    if (BRANCH_NAME == BRANCH_DEV) {
+                        env.DEPLOY_NAME = DEPLOY_DEV
+                        sh "cp api/target/universal/${ORGANIZATION}-${MODULE_NAME}-*.zip ${DEPLOY_DEV}-${ORGANIZATION}-${MODULE_NAME}.zip"
+                    }
+                    else if  (BRANCH_NAME == BRANCH_TEST) {
+                        env.DEPLOY_NAME = DEPLOY_TEST
+                        sh "cp api/target/universal/${ORGANIZATION}-${MODULE_NAME}-*.zip ${DEPLOY_TEST}-${ORGANIZATION}-${MODULE_NAME}.zip"
+                    }
+                    else if (BRANCH_NAME == BRANCH_PROD) {
+                        env.DEPLOY_NAME = DEPLOY_PROD
+                        sh "cp api/target/universal/${ORGANIZATION}-${MODULE_NAME}-*.zip ${DEPLOY_PROD}-${ORGANIZATION}-${MODULE_NAME}.zip"
+                    }
+                    else {
+                        colourText("info", "Not a deployable Git banch!")
+                    }
+                }
+            }
+        }
+
+        stage('Static Analysis') {
+            agent any
+            steps {
+                parallel (
+                        "Unit" :  {
+                            colourText("info","Running unit tests")
+                            // sh "$SBT test"
+                        },
+                        "Style" : {
+                            colourText("info","Running style tests")
+//                            sh """
+//                                $SBT scalastyleGenerateConfig
```

Comment:
Not great that this stuff is commented but hey, we'll soon be bringing in the sbt-code-quality plugin which will change the way the code quality tasks will be invoked.